### DISCUSSION TITLE
definePageMeta in options-api

### DISCUSSION THREAD

**Discussion by qlash** [nuxt/nuxt#29065] [Questions]
I need to have 'from' url on one of my pages. I managed to get it by options api:

```
<script lang="ts">
let fromName = '';

export default {
  beforeRouteEnter(_to, from, next) {
    fromName = from.name?.toString() ?? '';

    next();
  },
};
</script>
```
  
I also want to have definePageMeta set on that page, but when I use 

```
<script setup lang="ts">
definePageMeta({
  admin: true
});
```

It doesn't work. Is there a possibility to set route-meta tags in opitons api next to beforeRouteEnter?

**Comment by FelipeO16** [comment]
you can use [middleware](https://nuxt.com/docs/guide/directory-structure/middleware)

**Reply by qlash** [reply]
but how to pass a variable from middleware to the component?

**Reply by FelipeO16** [reply]
from store if you want, e.g: 
![image](https://github.com/user-attachments/assets/1a89da23-6057-4042-a7e4-1364db8d6948)
